server/src/main.rs

```diff
@@ -121,3 +121,105 @@ fn parse_args<'a, 'b>() -> clap::App<'a, 'b> {
                 .help("path where the different repositories are located"),
         )
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_web::test::TestServer;
+
+    #[test]
+    #[should_panic]
+    fn run_server_with_invalid_host() {
+        run_server("", DEFAULT_PORT, Path::new(DEFAULT_REPO_ROOT));
+    }
+
+    #[test]
+    #[should_panic]
+    fn run_server_with_invalid_port() {
+        run_server(DEFAULT_HOST, "", Path::new(DEFAULT_REPO_ROOT));
+    }
+
+    #[test]
+    #[should_panic]
+    fn run_server_with_invalid_repo() {
+        run_server(DEFAULT_HOST, DEFAULT_PORT, Path::new(""));
+    }
+
+    fn test_server() -> TestServer {
+        TestServer::build_with_state(|| {
+            let addr = GitRepos::new(git::load_repos(Path::new("../../"))).start();
+            AppState {
+                git_repos: addr.clone(),
+            }
+        })
+        .start(|app| {
+            app.resource("/repo/{repo}", |r| r.h(&super::get_repo));
+        })
+    }
+
+    #[test]
+    fn get_repo_with_file_in_root_folder() {
+        let mut srv = test_server();
+
+        let query = format!("/repo/{}?file={}&reference={}", "", "README.md", "heads/master");
```

Comment:
These are all interpolating static strings into another static string — best to just have one static string in thef first place?

```suggestion
        let query = "/repo/?file=README.md&reference=heads/master";
```

Reply:
Same for all the other `format!` calls in here.